src/lib/y2network/network_manager/connection_config_writers/vlan.rb

```diff
@@ -28,7 +28,7 @@ class Vlan < Base
         # @see Y2Network::ConnectionConfigWriters::Base#update_file
         # @param conn [Y2Network::ConnectionConfig::Vlan] Configuration to write
         def update_file(conn)
-          file.vlan["id"] = conn.vlan_id
+          file.vlan["id"] = conn.vlan_id.to_s
```

Comment:
Would be nice to add a unit test for VLAN connection which currently is missing ;)